website/docs/r/machine.html.markdown

```diff
@@ -9,3 +9,135 @@ description: |-
+  * `content` - A valid cloud config data in json-escaped yaml syntax
+
+* `constraints` - (Optional) Constraints that are used to drive placement policies for the virtual machine that is produced from this specification. Constraint expressions are matched against tags on existing placement targets.
+example:[{"mandatory" : "true", "expression": "environment":"prod"}, {"mandatory" : "false", "expression": "pci"}]. It is nested argument with the following properties.
+  * `expression` - A constraint that is conveyed to the policy engine. An expression of the form "[!]tag-key[:[tag-value]]", used to indicate a constraint match on keys and values of tags. 
+  * `mandatory` - Indicates whether this constraint should be strictly enforced or not.
+
+* `custom_properties` - (Optional) Additional custom properties that may be used to extend the machine.
+
+* `description` - (Optional) Describes machine within the scope of your organization and is not propagated to the cloud.
+
+* `disks` - (Optional) A set of disk specifications for this machine. It is nested argument with the following properties.
+  * `block_device_id` - The id of the existing block device.
+  * `description` - A human-friendly description.
+  * `name` - A human-friendly name used as an identifier in APIs that support this option.
+
+* `image_disk_constraints` - (Optional) Constraints that are used to drive placement policies for the image disk. Constraint expressions are matched against tags on existing placement targets. example:[{"mandatory" : "true", "expression": "environment:prod"}, {"mandatory" : "false", "expression": "pci"}]. It is nested argument with the following properties.
+  * `expression` - A constraint that is conveyed to the policy engine. An expression of the form "[!]tag-key[:[tag-value]]", used to indicate a constraint match on keys and values of tags. 
+  * `mandatory` - Indicates whether this constraint should be strictly enforced or not.
+
+* `image_ref` - (Optional) Direct image reference used for this machine (name, path, location, uri, etc.). Valid if no image property is provided
+
+* `nics` - (Optional) Specification for attaching nic to machine. A set of network interface controller specifications for this machine. If not specified, then a default network connection will be created. It is nested argument with the following properties.
+  * `addresses` - A list of IP addresses allocated or in use by this network interface.
+  * `customProperties` - Additional properties that may be used to extend the base type.
+  * `securityGroupIds` - list of security group ids which this network interface will be assigned to.
+  * `name` - A human-friendly name used as an identifier in APIs that support this option.
+  * `description` - A human-friendly description.
+  * `networkId` - Id of the network instance that this network interface plugs into.
+  * `deviceIndex` - The device index of this network interface.
```

Comment:
Please specify which of these properties are optional vs. required, if nics is provided in the config.

Reply:
Done